abrams_strogatz/lang_comp_bilinguals.py

```diff
@@ -0,0 +1,300 @@
+import matplotlib as mpl
+import matplotlib.pyplot as plt
+import numpy as np
+
+# Initial condition. Randomly distributed speakers.
+
+
+def initial_cond(m: int, n: int, pa: float = 1./3., pb: float = 1./3.) -> np.array:
+    """Creates the initial (mxn) array.
+
+    Each point within the (mxn) array represents a citizen speaking
+    language A, language B, or language A and B. The default initial 
+    condition corresponds to a scenario where the speakers are 
+    randomly distributed, the probability of each speaker being 
+    p(A) = pa, p(B) = pb, and p(AB) = 1. - pa - pb.
+
+    Args:
+        m: Size of the array.
+        n: Size of the array.
+        pa: Probability that a single node within the array speaks
+            language A. Defaults to 0.33.
+        pb: Probability that a single node within the array speaks
+            language B. Defaults to 0.33. 
+
+    Returns:
+        Returns a np.array(shape=(m, n)) where each node speaks either
+        language A (represented by a value 1), language B (represented
+        by a value -1) or laguage A and B (represented by a value 0). 
+        The latter represent bilingual speakers. 
+
+    """
+    popu = np.random.choice([1, 0, -1], size=(m, n), p=[pa, 1.0 - pa - pb, pb])
+    return popu
+
+
+def repres(popu: np.array) -> plt.figure:
+    """Array representation.
+
+    Graphical 2D-representation of the (mxn) array. Each site represents
+    an individual speaking either language A, language B, or language A 
+    and B (bilinguals). Languages are pictured by a selection of colors 
+    (blue, white, red).
+
+    Args:
+        popu: Array containing the language spoken by the population.
+            The values contained inside the array are 1 (lang A), 
+            0 (bilingual) and -1 (lang B).
+
+    Returns:
+        Returns a mpl.figure object containing the graphical representation
+            of the array.
+
+    """
+    col = mpl.colors.ListedColormap(["Blue", "White", "Red"])
+    colbar_tick = np.array([-1, 0, 1])
+    fig = plt.figure()
+    ax = plt.axes()
+    plot = ax.matshow(popu, cmap=col, origin="lower", animated=True)
+    ax.xaxis.tick_bottom()
+    ax.xaxis.set_major_locator(plt.MultipleLocator(2))
+    ax.yaxis.set_major_locator(plt.MultipleLocator(2))
+    ax.xaxis.set_major_formatter(lambda val, pos: r"{}".format(int(val) + 1))
+    ax.yaxis.set_major_formatter(lambda val, pos: r"{}".format(int(val) + 1))
+    fig.colorbar(plot, ax=ax, ticks=colbar_tick, label="Language").ax.set_yticklabels(
+        ["B", "AB", "A"])
```

Comment:
261:  81 C815 missing trailing comma in Python 3.5+ [flake8-commas]
```suggestion
        ["B", "AB", "A"],)
```